test/integration/link_dom.cc

```diff
@@ -358,6 +358,31 @@ TEST(DOMLink, Sensors)
   ASSERT_NE(nullptr, thermalCamSensor);
   EXPECT_EQ("my_thermal_camera", thermalCamSensor->Name());
 
+  // Get the segmentation sensor
+  const sdf::Sensor *segmentationSensor =
+    link->SensorByName("segmentation_sensor");
+  ASSERT_NE(nullptr, segmentationSensor);
+  EXPECT_EQ("segmentation_sensor", segmentationSensor->Name());
+  EXPECT_EQ(sdf::SensorType::SEGMENTATION_CAMERA, segmentationSensor->Type());
+  EXPECT_EQ(ignition::math::Pose3d(37, 38, 39, 0, 0, 0),
+            segmentationSensor->RawPose());
+  const sdf::Camera *segmentationCameraSensor =
+    segmentationSensor->CameraSensor();
+  ASSERT_NE(nullptr, segmentationCameraSensor);
+  EXPECT_EQ("my_segmentation_camera", segmentationCameraSensor->Name());
+
+  // Get the boundingbox sensor
+  const sdf::Sensor *boundingboxSensor =
+    link->SensorByName("boundingbox_sensor");
+  ASSERT_NE(nullptr, boundingboxSensor);
+  EXPECT_EQ("boundingbox_sensor", boundingboxSensor->Name());
+  EXPECT_EQ(sdf::SensorType::BOUNDINGBOX_CAMERA, boundingboxSensor->Type());
+  EXPECT_EQ(ignition::math::Pose3d(37, 38, 39, 0, 0, 0),
+            boundingboxSensor->RawPose());
+  const sdf::Camera *boundingboxCamSensor = boundingboxSensor->CameraSensor();
+  ASSERT_NE(nullptr, boundingboxCamSensor);
+  EXPECT_EQ("my_boundingbox_camera", boundingboxCamSensor->Name());
```

Comment:
```suggestion
  EXPECT_EQ("my_boundingbox_camera", boundingboxCamSensor->Name());
  EXPECT_TRUE(boundingboxCamSensor->HasBoundingBoxType());
  EXPECT_EQ("2d", boundingboxCamSensor->BoundingBoxType());
```
Now that the bbox type is a part of the sdf::Camera API, can you add a check for the camera's bbox type?